### PR TITLE
Remove cuDF dependency from pylibcudf column from_device tests

### PR DESCRIPTION
**Pull request by brandon-b-miller**
This removes the need to `import cudf` in `test_column_from_device` and removes a runtime dependency on numpy in the associated pylibcudf column method.

### CODE REVIEW

File: python/cudf/cudf/pylibcudf_tests/test_column_from_device.py [hunk 1]
```diff
@@ -35,17 +36,39 @@ def valid_type(request):
     return request.param
 
 
+class DataBuffer:
```
Comment (wence-) [inline python/cudf/cudf/pylibcudf_tests/test_column_from_device.py:39]: Are we allowed to (in the tests) depend on cupy, then we could just use `cp.asarray()` in the test?

**Review by wence- — CHANGES_REQUESTED** [review]

File: python/cudf/cudf/pylibcudf_tests/test_column_from_device.py (hunk 1 above)
Comment (brandon-b-miller) [reply python/cudf/cudf/pylibcudf_tests/test_column_from_device.py:39]: Was discussing this with @lithomas1 , the issue with cupy is the missing support for datelike types

```
>>> cp.asarray([1,2,3], dtype='datetime64[ns]')
ValueError: Unsupported dtype datetime64[ns]
```

Comment (wence-) [reply python/cudf/cudf/pylibcudf_tests/test_column_from_device.py:39]: Ah, ok

**Review by lithomas1 — APPROVED** [review]
Thanks Brandon!

**Review by wence- — APPROVED** [review]

**Pull request merged**